colorbleed/plugins/maya/publish/collect_look.py

```diff
@@ -89,20 +86,25 @@ def collect(self, instance):
         self.log.info("Gathering set relations..")
         for objset in sets:
             self.log.debug("From %s.." % objset)
-            content = cmds.sets(objset, query=True)
             objset_members = sets[objset]["members"]
-            for member in cmds.ls(content, long=True):
+
+            # Get all nodes of the current objectSet
+            for member in cmds.ls(cmds.sets(objset, query=True), long=True):
                 member_data = self.collect_member_data(member,
                                                        objset_members,
-                                                       instance_lookup,
-                                                       verbose)
+                                                       instance_lookup)
                 if not member_data:
                     continue
 
+                # Add information of the node to the members list
                 sets[objset]["members"].append(member_data)
 
-        # Remove sets that didn't have any members assigned in the end
-        sets = self.remove_sets_without_members(sets)
+            # Remove sets that didn't have any members assigned in the end
+            # Thus the data will be limited to only what we need.
+            if not sets[objset]["members"]:
```

Comment:
Great! Much easier to my eyes without needing to go to a separate function. :)

The only issue here is you can't pop the key from the dictionary as you are currently iterating over it. Otherwise it would behave similar to a list iterator where it then suddenly skips an index. In Python 2.7 this will even raise an error on dictionaries:

```python
a = {"a": 1, "b": 2, "c": 3}
for key in a:
    a.pop(key)
# # RuntimeError: dictionary changed size during iteration # 
```

A solution would to iterate over `a.keys()` (which makes a copy of the keys) instead of the `dict` itself. :) Though, I'm not sure how that works in Python 3+ because `dict.keys()` there I believe also returns an iterator. Feel free to test there too and find a behavior that works.

Reply:
I've tested this, even with `dict.keys()` it will not work in Python 3 because it's an iterator, just like iterating over the dictionary itself.

The solution is to explicitly turn it into a copy of the keys using `list`, e.g.
```python
a = {"a": 1, "b": 2, "c": 3}
for key in list(a):
    a.pop(key)
```
This will work in both python 2 and 3.